services/sensiolabs/sensiolabs.service.js

```diff
@@ -1,110 +1,119 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
+const Joi = require('joi')
+const BaseXmlService = require('../base-xml')
 const serverSecrets = require('../../lib/server-secrets')
+const { statusRegex } = require('./sensiolabs-helpers')
+
+const schema = Joi.object({
+  project: Joi.object({
+    'last-analysis': Joi.object({
+      status: Joi.string()
+        .regex(statusRegex)
+        .required(),
+      grade: Joi.string()
+        .regex(/platinum|gold|silver|bronze|none/)
+        .allow('', null),
+    }),
+  }).required(),
+}).required()
+
+module.exports = class Sensiolabs extends BaseXmlService {
+  static render({ status, grade }) {
+    if (status !== 'finished') {
+      return {
+        message: 'pending',
+        color: 'grey',
+      }
+    }
+    let color,
+      message = grade
+    if (grade === 'platinum') {
+      color = 'brightgreen'
+    } else if (grade === 'gold') {
+      color = 'yellow'
+    } else if (grade === 'silver') {
+      color = 'lightgrey'
+    } else if (grade === 'bronze') {
+      color = 'orange'
+    } else {
+      message = 'no medal'
+      color = 'red'
+    }
+
+    return {
+      message,
+      color,
+    }
+  }
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class Sensiolabs extends LegacyService {
   static get category() {
     return 'build'
```

Comment:
IMO `quality` is a better fitting category, although I wasn't sure if changing the category as part of a refactor was okay?

Reply:
Yes, I think it makes sense to fix things like that as services are refactored.